pkg/vm/vm.go

```diff
@@ -1147,29 +1143,27 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 
 		elem := v.estack.Pop()
 		switch t := elem.value.(type) {
-		case *ArrayItem:
-			a := t.value
+		case *stackitem.Array:
+			a := t.Value().([]stackitem.Item)
 			k := int(key.BigInt().Int64())
 			if k < 0 || k >= len(a) {
 				panic("REMOVE: invalid index")
 			}
 			v.refs.Remove(a[k])
-			a = append(a[:k], a[k+1:]...)
-			t.value = a
-		case *StructItem:
-			a := t.value
+			t.Remove(k)
+		case *stackitem.Struct:
+			a := t.Value().([]stackitem.Item)
 			k := int(key.BigInt().Int64())
 			if k < 0 || k >= len(a) {
 				panic("REMOVE: invalid index")
 			}
 			v.refs.Remove(a[k])
-			a = append(a[:k], a[k+1:]...)
-			t.value = a
-		case *MapItem:
+			t.Remove(k)
+		case *stackitem.Map:
 			index := t.Index(key.Item())
 			// NEO 2.0 doesn't error on missing key.
```

Comment:
BTW, is this still relevant for Neo 3?

Reply:
It is, so I updated this comment (see references in commit comment).

Reply:
https://github.com/neo-project/neo-vm/blob/9d02720044d161c2a334d8e5a66130d3feda4066/src/neo-vm/Types/Map.cs#L77
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.remove?view=netframework-4.8#System_Collections_Generic_IDictionary_2_Remove__0_
https://github.com/neo-project/neo-vm/blob/9d02720044d161c2a334d8e5a66130d3feda4066/src/neo-vm/ExecutionEngine.cs#L1244

I think this needs to be fixed (and tested), maybe in a separate PR.

Reply:
Ok, for now I removed this commit. Let's end up with this PR and I''ll create a separate one for this issue.